tests/functional/report/monitoring.js

```diff
@@ -0,0 +1,106 @@
+'use strict'; // eslint-disable-line strict
+const http = require('http');
+const assert = require('assert');
+
+describe('Monitoring - getting metrics', () => {
+    const conf = require('../config.json');
+
+    async function query(path, method = 'GET', token = 'report-token-1') {
+        return new Promise(resolve => http.request({
+            method,
+            host: conf.ipAddress,
+            path,
+            port: 8000,
+            headers: { 'x-scal-report-token': token },
+        }, () => resolve()).end());
+    }
+
+    async function getMetrics() {
+        return new Promise(resolve => {
+            http.get({ host: conf.ipAddress, path: '/metrics', port: 8002 }, res => {
+                assert.strictEqual(res.statusCode, 200);
+
+                let body = '';
+                res.on('data', chunk => { body += chunk; });
+                res.on('end', () => resolve(body));
```

Comment:
Does not really matter much for tests but I recommend using an array and push, with a join in end. Slightly more efficient. We also use this pattern a lot so it would be consistent. 

```suggestion
                let body = [];
                res.on('data', chunk => { body.push(chunk); });
                res.on('end', () => resolve(body.join('')));
```